src/Fantomas.Client/FantomasToolLocator.fs

```diff
@@ -141,33 +133,124 @@ let private (|CompatibleTool|_|) lines =
+
+    fantomasExecutableOnPathOpt
+    |> Option.bind
+        (fun fantomasExecutablePath ->
+            let processStart = ProcessStartInfo(fantomasExecutablePath)
+            processStart.Arguments <- "--version"
+            processStart.RedirectStandardOutput <- true
+            processStart.CreateNoWindow <- true
+            processStart.RedirectStandardOutput <- true
+            processStart.RedirectStandardError <- true
+            processStart.UseShellExecute <- false
+
+            match startProcess processStart with
+            | Ok p ->
+                p.WaitForExit()
+                let stdOut = p.StandardOutput.ReadToEnd()
+
+                stdOut
+                |> Option.ofObj
+                |> Option.map
+                    (fun s ->
+                        let version =
+                            s
+                                .ToLowerInvariant()
+                                .Replace("fantomas", String.Empty)
+                                .Trim()
+
+                        FantomasExecutableFile(fantomasExecutablePath), FantomasVersion(version))
+            | Error (ProcessStartError.ExecutableFileNotFound _)
+            | Error (ProcessStartError.UnExpectedException _) -> None)
+
+let findFantomasTool (workingDir: Folder) : Result<FantomasToolFound, FantomasToolError> =
+    // First try and find a local tool for the folder.
+    // Next see if there is a global tool.
+    // Lastly check if an executable `fantomas` is present on the PATH.
+    let localToolsListResult = runToolListCmd workingDir false
+
+    match localToolsListResult with
+    | Ok (CompatibleTool version) -> Ok(FantomasToolFound(version, FantomasToolStartInfo.LocalTool workingDir))
+    | Error err -> Error(FantomasToolError.DotNetListError err)
+    | Ok _localToolListResult ->
+        let globalToolsListResult = runToolListCmd workingDir true
+
+        match globalToolsListResult with
+        | Ok (CompatibleTool version) -> Ok(FantomasToolFound(version, FantomasToolStartInfo.GlobalTool))
+        | Error err -> Error(FantomasToolError.DotNetListError err)
+        | Ok _nonCompatibleGlobalVersion ->
+            let fantomasOnPathVersion = fantomasVersionOnPath ()
+
+            match fantomasOnPathVersion with
+            | Some (executableFile, FantomasVersion (CompatibleVersion version)) ->
+                Ok(FantomasToolFound((FantomasVersion(version)), FantomasToolStartInfo.ToolOnPath executableFile))
+            | _ -> Error FantomasToolError.NoCompatibleVersionFound
+
+let createFor (startInfo: FantomasToolStartInfo) : Result<RunningFantomasTool, ProcessStartError> =
+    let processStart =
+        match startInfo with
+        | FantomasToolStartInfo.LocalTool (Folder workingDirectory) ->
+            let ps = ProcessStartInfo("dotnet")
+            ps.WorkingDirectory <- workingDirectory
+            ps.Arguments <- "fantomas --daemon"
+            ps
+        | FantomasToolStartInfo.GlobalTool ->
```

Comment:
in both of these new cases I think the working directory needs to be set as well, doesn't it? since the daemon will probe for configs based on that file structure?

Reply:
Actual both `.editorconfig` and `.fantomasignore` are detected from the file's point of view.
See https://github.com/fsprojects/fantomas/blob/4.6/src/Fantomas.Extras/IgnoreFile.fs
For `.editorconfig` that just works out of the box as the spec is based on all of this.

Reply:
oh great 👍